xonsh/foreign_shells.py

```diff
@@ -540,9 +541,17 @@ def load_foreign_aliases(shells=None, config=None, issue_warning=True):
     """
     shells = _get_shells(shells=shells, config=config, issue_warning=issue_warning)
     aliases = {}
+    xonsh_aliases = builtins.aliases
     for shell in shells:
         shell = ensure_shell(shell)
         _, shaliases = foreign_shell_data(**shell)
-        if shaliases:
-            aliases.update(shaliases)
+        if not shaliases:
+            continue
```

Comment:
This doesn't block the PR, just an opportunity to get rid of a couple more lines :)